Fix for ffs() to correctly handle the case when it is called with argument of
zero.
This patch by Jurij Smakov <jurij@wooyd.org>
Submitted upstream to X.Org:
https://bugs.freedesktop.org/show_bug.cgi?id=8968

Index: xorg-server/dix/ffs.c
===================================================================
--- xorg-server.orig/dix/ffs.c	2006-10-21 21:44:19.000000000 +0200
+++ xorg-server/dix/ffs.c	2006-11-10 21:55:33.000000000 +0100
@@ -31,6 +31,7 @@
 ffs(int i)
 {
     int j;
+    if (i == 0) return 0;
     for (j = 1; (i & 1) == 0; j++)
 	i >>= 1;
     return j;
